pkg/types/resources.go

```diff
@@ -57,6 +57,44 @@ type ClusterStatus struct {
 	Phase string
 }
 
+type CmkTemplate struct {
```

Comment:
Curious about all these structs
Do they need to be public? And do they need to be here? They seem very much tied to the cloudstack api, so shouldn't they be encapsulated with cmk?

Reply:
They are tied to the cloudstack API. I thought I recalled seeing some vsphere structs in here and thought it would be fitting to add cloudstack structs here as well. I'll find somewhere else to put them